src/goto-programs/goto_program.h

```diff
@@ -435,17 +435,17 @@ class goto_programt
 
     /// Constructor that can set all members
     instructiont(
-      codet &&_code,
+      const codet &_code,
```

Comment:
Why revert that recent change? Especially with the changes below, which would be very amenable to rvalue references?

Reply:
The best approach might be to pass all these by value (and then use `std::move` in constructing the individual members) so that necessary copies (such as a `const source_locationt`) don't stop the compiler from using this constructor.

Reply:
Indeed, will tweak